ordered_product/tests.py

```diff
@@ -0,0 +1,88 @@
+import pytest
+from ordered_product.models import OrderedProduct
+from django.core.exceptions import ValidationError
+
+
+@pytest.fixture
+def save_objects(delivery_location0, supplier0, supplier_product0, client0, product0):
+    client0.save()
+    product0.save()
+    supplier0.save()
+    supplier_product0.save()
+    delivery_location0.save()
+
+
+class TestOrderedProductModel():
+    @pytest.mark.django_db()
+    def test_save_and_delete_ordered_product(self, ordered_product0, save_objects):
+        assert ordered_product0 not in OrderedProduct.objects.all()
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 in OrderedProduct.objects.all()
+        ordered_product0.delete_ordered_product()
+        assert ordered_product0 not in OrderedProduct.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_user_name(self, ordered_product0, save_objects, client1, client0):
+        ordered_product0.user_name = client0
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_user_name(client1))
+        assert ordered_product0 in list(OrderedProduct.filter_user_name(client0))
+
+    @pytest.mark.django_db()
+    def test_filter_by_supplier_product_id(self, ordered_product0, supplier_product0, supplier_product1, save_objects):
+        ordered_product0.supplier_product_id = supplier_product0
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_supplier_product_id(supplier_product1))
+        assert ordered_product0 in list(OrderedProduct.filter_supplier_product_id(supplier_product0))
+
+    @pytest.mark.django_db()
+    def test_filter_delivery_location_id(self, ordered_product0, delivery_location0, delivery_location1, save_objects):
+        ordered_product0.delivery_location_id = delivery_location0
+        ordered_product0.save_ordered_product()
+        assert ordered_product0 not in list(OrderedProduct.filter_delivery_location_id(delivery_location1))
+        assert ordered_product0 in list(OrderedProduct.filter_delivery_location_id(delivery_location0))
+
+    @pytest.mark.django_db()
+    def test_validators(self, delivery_location0, client0, supplier_product0, save_objects):
+        with pytest.raises(ValidationError):
+            temp_ordered_product = OrderedProduct(
+                delivery_location_id=delivery_location0,
+                user_name=client0,
+                supplier_product_id=supplier_product0,
+                quantity=-2)
+            temp_ordered_product.save_ordered_product()
+        assert temp_ordered_product not in OrderedProduct.objects.all()
```

Comment:
make the context manager include only the line that you expect to generate the exception. This way you increase the chances for actually spotting the error you were expecting.
```suggestion
        temp_ordered_product = OrderedProduct(
                delivery_location_id=delivery_location0,
                user_name=client0,
                supplier_product_id=supplier_product0,
                quantity=-2)
        with pytest.raises(ValidationError):
            temp_ordered_product.save_ordered_product()
        assert temp_ordered_product not in OrderedProduct.objects.all()
```